apps/src/templates/progress/googlePlatformApiRedux.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Google Classroom Share Button is only available through the google
+ * platform api, so we have to add it to our page on load. We manage loading
+ * in redux since multiple components need to observe the loading state.
+ */
+
+const START_LOADING_GAPI = 'googlePlatformApi/START_LOADING_GAPI';
+const FINISH_LOADING_GAPI = 'googlePlatformApi/FINISH_LOADING_GAPI';
+
+const startLoadingGapi = startTime => ({type: START_LOADING_GAPI, startTime});
+const finishLoadingGapi = success => ({type: FINISH_LOADING_GAPI, success});
+
+const GOOGLE_PLATFORM_API_ID = 'GooglePlatformApiId';
+const LOAD_TIMEOUT_MILLIS = 15000;
+
+const initialState = {
+  loading: false,
+  loaded: false,
+  loadStartTime: null
+};
+
+export function loadGooglePlatformApi() {
+  return (dispatch, getState) => {
+    if (gapiReady()) {
```

Comment:
A test for this logic would be awesome!

Reply:
do you mean a test to make sure that if the api is already loaded we don't load it again?

Reply:
actually now that i think about it i have no idea how to test this module at all. can we hop on a call to discuss tomorrow?

Reply:
Haha sure. To be completely honest, part of the reason I asked for a test is I'm not really sure how to test functions that use this structure - I was hoping you had seen some docs about it while you were investigating. But I can definitely help with general redux testing (mocking the store, mocking various functions, etc).